test/hns_test.go

```diff
@@ -0,0 +1,110 @@
+package hcsshimtest
+
+import (
+	"os"
+	"testing"
+
+	"github.com/microsoft/hcsshim"
+)
+
+const (
+	NatTestNetworkName  string = "GoTestNat"
+	NatTestEndpointName string = "GoTestNatEndpoint"
+)
+
+func TestMain(m *testing.M) {
+	os.Exit(m.Run())
+}
+
+func TestNetworkCreate(t *testing.T) {
+	network := &hcsshim.HNSNetwork{
+		Type: "NAT",
+		Name: NatTestNetworkName,
+		Subnets: []hcsshim.Subnet{
+			hcsshim.Subnet{
+				AddressPrefix:  "192.168.100.0/24",
+				GatewayAddress: "192.168.100.1",
+			},
+		},
+	}
+
+	_, err := network.Create()
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestEndpointCreate(t *testing.T) {
+	network, err := hcsshim.GetHNSNetworkByName(NatTestNetworkName)
```

Comment:
Golang tests are not guaranteed to run in order (or all run). Each of these tests needs to be self sufficient and cleanup after themselves or be combined to be one large test.

That said, this seems to mostly be testing that the platform code doesn't error out, should it be here? We didn't add container tests to hcsshim because of the platform dependencies for testing. I'm not against the tests, but I'm not sure I see value in them here.